Automattic/jetpack
# Update to eslint 9

## Proposed changes:

Eslint 9 brings a new config format, which means this PR also rewrites all the eslint configs. I took the opportunity to clean them up (although a good bit of that wound up in earlier PRs), and added code to fetch the textdomain from composer.json and whether the project uses React from package.json so we don't need to have so many boilerplate configs that just set a textdomain or enable React rules.

### Other information:


## Jetpack product discussion


Draft announcement: pdWQjU-14g-p2

## Does this pull request change what data or activity we track or use?


None

## Testing instructions:






* CI happy?
* Changes make sense?
* Double check that the built artifacts don't suddenly start including unneeded files like `eslint.config.mjs`.
* If you really want to dig into things, verify that all the same files are being linted as were before, and check that React rules are being enabled for all the React files, and so on. 😀
